src/vm_cpu.rs

```diff
@@ -552,8 +542,9 @@ impl VmCpu {
         let mut sstatus = LocalRegisterCopy::<u64, sstatus::Register>::new(0);
         sstatus.modify(sstatus::spp::Supervisor);
         sstatus.modify(sstatus::fs::Initial);
-        #[cfg(target_feature = "v")]
-        sstatus.modify(sstatus::vs::Initial);
+        if *crate::VECTORS_ENABLED.call_once(|| panic!("Unsure of vectors state")) {
+            sstatus.modify(sstatus::vs::Initial);
+        }
```

Comment:
See my previous comment -- I think using CpuInfo here would be much simpler.

Reply:
see above